spec/unit/knife_spec.rb

```diff
@@ -374,6 +375,16 @@ class SuperAwesomeCommand < Chef::Knife
       expect(stderr.string).to match(%r[Response: nothing to see here])
     end
 
+    it "formats 406s (non-supported API version error) nicely" do
+      response = Net::HTTPNotAcceptable.new("1.1", "406", "Not Acceptable")
+      response.instance_variable_set(:@read, true) # I hate you, net/http.
+      allow(response).to receive(:body).and_return(Chef::JSONCompat.to_json(:error => "sad trombone", :min_version => "0", :max_version => "1"))
+      allow(knife).to receive(:run).and_raise(Net::HTTPServerException.new("406 Not Acceptable", response))
+      knife.run_with_pretty_exceptions
+      expect(stderr.string).to match(%r[The version of Chef that Knife is using is not supported by the Chef server you sent this request to])
+      expect(stderr.string).to match(%r[This version of Chef requires a server API version of #{Chef::HTTP::Authenticator::SERVER_API_VERSION}])
```

Comment:
When you don't need any regex features, it's better to do

```ruby
expect(actual_string).to include("expected part of string")
```

Just so you don't stub your toe on any special regex characters (or have to worry about escaping them).